### PR TITLE
added new message alerting

### PR DESCRIPTION
**Pull request by DirtDeodara**
There was a big auto reformat that happened in shoutout.jsx. im sorry about that.

### CODE REVIEW

File: src/components/Shoutout/Shoutout.jsx [hunk 1]
```diff
@@ -1,150 +1,212 @@
...
+        aria-hidden={"false"}
+      >
+        {symbol}
+      </span>
+      <span>{emojiCount > 0 && emojiCount}</span>
+    </div>
+  );
+};
+
+const CommentContainer = ({ color, id, open, setHasNewComments }) => {
```
Comment (DirtDeodara) [inline src/components/Shoutout/Shoutout.jsx:61]: passed in setHasNewComments

File: src/components/Shoutout/Shoutout.jsx [hunk 2]
```diff
@@ -1,150 +1,212 @@
...
+    return storedCommentList !== null ? JSON.parse(storedCommentList) : [];
   });
 
   useEffect(() => {
-    socket.on('comment', comment => {
+    socket.on("comment", (comment) => {
       if (id === comment.id) {
-        setComments([...comments, comment])
+        setComments([...comments, comment]);
+        setHasNewComments(true)
```
Comment (DirtDeodara) [inline src/components/Shoutout/Shoutout.jsx:72]: modifying state if there was a new comment pushed to the array

File: src/components/Shoutout/Shoutout.jsx [hunk 3]
```diff
@@ -1,150 +1,212 @@
...
+          color: colorFactory[color].text,
+        }}
+      >
         <div className="shoutoutContainerPadding">
           <CommentIcon
             alt="comment"
-            className={`${commentsOpen ? "commentIconOpen" : "commentIconClosed"} commentIcon`}
+            className={`${
+              commentsOpen ? "commentIconOpen" : "commentIconClosed"
+            } ${hasNewComments && !hasBeenRead ? "pulse" : null } commentIcon`}
```
Comment (DirtDeodara) [inline src/components/Shoutout/Shoutout.jsx:155]: adding classname if there are new comments and the message component was closed

File: src/components/Shoutout/Shoutout.jsx [hunk 4]
```diff
@@ -1,150 +1,212 @@
...
-  text,
-  handleConfetti
-}) => {
+const Shoutout = ({ author, color, id, recipient, text, handleConfetti }) => {
   const [commentsOpen, setCommentsOpen] = useState(false);
-  const toggleCommentsOpen = () => setCommentsOpen(!commentsOpen);
+  const [hasNewComments, setHasNewComments] = useState(false);
+  const [hasBeenRead, setHasBeenRead] = useState(false);
+
+  const toggleCommentsOpen = () => {
```
Comment (DirtDeodara) [inline src/components/Shoutout/Shoutout.jsx:127]: when the comments get opened the state changes to stop the pulsing

File: src/components/Shoutout/Shoutout.jsx [hunk 5]
```diff
@@ -1,150 +1,212 @@
...
+    setCommentsOpen(!commentsOpen);
+    setHasBeenRead(true);
+    setHasNewComments(false);
+  };
+
+  useEffect(() => {
```
Comment (DirtDeodara) [inline src/components/Shoutout/Shoutout.jsx:133]: prevents the pulse if the comments are open

**Review by jennagoldman — APPROVED** [review]

**Review by DannieSchu — APPROVED** [review]

**Pull request merged**